File: internal/model/host.go
package model

import (
	"gorm.io/gorm"
)

// Host 主机模型
type Host struct {
	gorm.Model
	Name        string `json:"name" gorm:"not null"`
	IP          string `json:"ip" gorm:"not null"`
	Port        int    `json:"port" gorm:"not null"`
	Username    string `json:"username" gorm:"not null"`
	Password    string `json:"password"`
	SSHKey      string `json:"ssh_key"`
	Status      string `json:"status"`
	Description string `json:"description"`
}

// FileSync 文件同步任务模型
type FileSync struct {
	gorm.Model
	HostID        uint    `json:"host_id" gorm:"not null"`
	Host          Host    `json:"host" gorm:"foreignKey:HostID"`
	SourcePath    string  `json:"sourcePath" gorm:"not null"`
	TargetPath    string  `json:"targetPath" gorm:"not null"`
	Status        string  `json:"status"`      // syncing, paused, completed, failed, cancelled
	Progress      float64 `json:"progress"`    // 同步进度 0-100
	Speed         float64 `json:"speed"`       // 传输速度 bytes/s
	LastSyncAt    string  `json:"last_sync_at"`
	Description   string  `json:"description"`
	MD5           string  `json:"md5"`
	ModifiedTime  int64   `json:"modified_time"`
	FileSize      int64   `json:"file_size"`
	IsIncremental bool    `json:"isIncremental"`
	SyncedSize    int64   `json:"synced_size"`  // 已同步大小，用于断点续传
	IsPaused      bool    `json:"is_paused"`    // 是否暂停
}

// FileSyncHistory 文件同步历史记录
type FileSyncHistory struct {
	gorm.Model
	FileSyncID   uint   `json:"file_sync_id" gorm:"not null"`
	FileSync     FileSync `json:"file_sync" gorm:"foreignKey:FileSyncID"`
	Status       string `json:"status"`
	Message      string `json:"message"`
	MD5          string `json:"md5"`
	FileSize     int64  `json:"file_size"`
	SyncType     string `json:"sync_type"` // full: 全量同步, incremental: 增量同步
} 

File: web/src/api/host.ts
import request from '../utils/request';
import { ApiResponse } from '../types/api';

export interface Host {
  id: number;
  name: string;
  ip: string;
  port: number;
  username: string;
  password?: string;
  sshKey?: string;
  status: string;
  description?: string;
}

export interface FileSync {
  id: number;
  hostId: number;
  sourcePath: string;
  targetPath: string;
  status: string;
  lastSyncAt?: string;
  description?: string;
  md5?: string;
  modifiedTime?: number;
  fileSize?: number;
  isIncremental?: boolean;
  progress?: number;    // 同步进度 0-100
  speed?: number;      // 传输速度 bytes/s
  syncedSize?: number; // 已同步大小
  isPaused?: boolean;  // 是否暂停
}

export interface FileSyncHistory {
  id: number;
  fileSyncId: number;
  status: string;
  message: string;
  md5: string;
  fileSize: number;
  syncType: string;
  createdAt: string;
}

export async function getHostList() {
  const response = await request.get<ApiResponse<Host[]>>('/api/v1/hosts/list');
  return response.data.data || [];
}

export async function createHost(data: Partial<Host>) {
  await request.post<ApiResponse<void>>('/api/v1/hosts/create', data);
}

export async function updateHost(id: number, data: Partial<Host>) {
  await request.put<ApiResponse<void>>(`/api/v1/hosts/${id}`, data);
}

export async function deleteHost(id: number) {
  await request.delete<ApiResponse<void>>(`/api/v1/hosts/${id}`);
}

export async function syncFile(data: Partial<FileSync>) {
  await request.post<ApiResponse<void>>('/api/v1/hosts/sync', data);
}

export async function getFileSyncs(hostId: number) {
  const response = await request.get<ApiResponse<FileSync[]>>(`/api/v1/hosts/${hostId}/syncs`);
  return response.data.data || [];
}

export async function getSyncHistory(fileSyncId: number) {
  const response = await request.get<ApiResponse<FileSyncHistory[]>>(`/api/v1/hosts/syncs/${fileSyncId}/history`);
  return response.data.data || [];
}

export async function pauseSync(fileSyncId: number) {
  await request.post<ApiResponse<void>>(`/api/v1/hosts/syncs/${fileSyncId}/pause`);
}

export async function resumeSync(fileSyncId: number) {
  await request.post<ApiResponse<void>>(`/api/v1/hosts/syncs/${fileSyncId}/resume`);
}

export async function cancelSync(fileSyncId: number) {
  await request.post<ApiResponse<void>>(`/api/v1/hosts/syncs/${fileSyncId}/cancel`);
} 